include/Dialect/PolyExt/IR/PolyExtOps.td

```diff
@@ -0,0 +1,43 @@
+#ifndef HEIR_INCLUDE_DIALECT_POLYEXT_IR_POLYEXTOPS_TD_
+#define HEIR_INCLUDE_DIALECT_POLYEXT_IR_POLYEXTOPS_TD_
+
+include "PolyExtDialect.td"
+include "include/Dialect/Poly/IR/PolyDialect.td"
+include "include/Dialect/Poly/IR/PolyTypes.td"
+include "mlir/IR/OpBase.td"
+include "mlir/Interfaces/InferTypeOpInterface.td"
+include "mlir/Interfaces/SideEffectInterfaces.td"
+
+
+class PolyExt_Op<string mnemonic, list<Trait> traits = []> :
+        Op<PolyExt_Dialect, mnemonic, traits> {
+  let cppNamespace = "::mlir::heir::poly_ext";
+}
+
+def PolyExt_CModSwitchOp : PolyExt_Op<"cmod_switch", traits = [Pure, ElementwiseMappable]> {
+  let summary = "Move polynomial into a ring with different `cmod` by scaling polynomial coefficients proportionally";
+
+  let description = [{
+    `poly_ext.cmod_switch` "moves" polinomial $x$ from ring $Z/QZ[X]/(f(x))$ to the ring $Z/qZ[X]/(f(x))$.
+    The move is done by scaling the polynomial coefficients by $q/Q$ and "rounding" so input and output coufficients are congruent modulo `congruence_modulus` $t$.
+    The resulting polynomial $\tilde{x} \approx (q/Q)x$ and $\tilde{x} \equiv x \mod t$.
+
+    To skip the "rounding" step, set `congruence_modulus` to 1.
+  }];
+  let arguments = (ins
+    PolynomialLike:$x,
+    // TODO: make congruence_modulus optional with default value 1
+    APIntAttr:$congruence_modulus
```

Comment:
Perhaps:

`DefaultValuedAttr<I64Attr, "1">` ? I don't think you can get away with an APInt attribute unless you define the storage class for it, and 64 bits is probably enough...

Reply:
There is a potential bug in ODS. I would work on repro of this in upstream but it may take some time. I would rather push it "as is" and address the TODO after the bug is fixed/denounced.

Reply:
I also ran into some difficulties trying to make defaults work, but never figured it out. OK to submit as is.